src/atoms/Button/Button.native.js

```diff
@@ -1,35 +1,315 @@
-
-const StyledButton = styled(TouchableOpacity)(
-  (props) => `
-  background-color: ${props.theme.colors.shade[900]};
-  padding: 14px;
-  border-radius: 6px;
-  margin-top: 7px;
-`,
-);
-
-const StyledText = styled(Text)(
-  (props) => `
-  color: ${props.theme.colors.tone[900]};
-  align-self: center;
-  font-size: 18px;
-  letter-spacing: ${props.theme.fonts.letterSpacing.l};
-`,
-);
-
-const Button = ({ onClick, children, ...rest }) => (
-  <StyledButton onPress={onClick} {...rest}>
-    <StyledText {...rest}>{children}</StyledText>
-  </StyledButton>
-);
+import styled, { ThemeContext } from 'styled-components/native';
+import { TouchableHighlight } from 'react-native';
+
+import Icon from '../Icon';
+import Space from '../Space';
+import Text from '../Text';
+import View from '../View';
+import Flex from '../Flex';
+import automation from '../../_helpers/automation-attributes';
+import { getColor, makePxValue, getVariantColorKeys } from '../../_helpers/theme';
+
+const styles = {
+  fontColor({ variant, variantColor, disabled }) {
+    switch (variant) {
+      case 'primary':
+        if (disabled) {
+          return 'light.150';
+        }
+        return 'light.100';
+      case 'secondary':
+        if (disabled) {
+          return 'shade.940';
+        }
+        return `${variantColor || 'primary'}.800`;
+      case 'tertiary':
+        if (disabled) {
+          return 'shade.940';
+        }
+        return `${variantColor || 'primary'}.800`;
+      default:
+        if (disabled) {
+          return 'light.150';
+        }
+        return 'light.100';
+    }
+  },
+  backgroundColor({ variant, variantColor, disabled, theme }) {
+    switch (variant) {
+      case 'primary':
+        if (disabled) {
+          return getColor(theme, `${variantColor || 'primary'}.500`);
+        }
+        return getColor(theme, `${variantColor || 'primary'}.800`);
+      case 'secondary':
+        if (disabled) {
+          return getColor(theme, 'shade.910');
+        }
+        return getColor(theme, `${variantColor || 'primary'}.920`);
+      case 'tertiary':
+        return 'transparent';
+      default:
+        if (disabled) {
+          return getColor(theme, `${variantColor || 'primary'}.500`);
+        }
+        return getColor(theme, `${variantColor || 'primary'}.800`);
+    }
+  },
+  border({ variant, variantColor, disabled, theme }) {
+    switch (variant) {
+      case 'primary':
+        return '0px';
+      case 'secondary':
+        if (disabled) {
+          return `${makePxValue(0.125)} solid ${getColor(theme, 'shade.930')}`;
+        }
+        return `${makePxValue(0.125)} solid ${getColor(theme, `${variantColor || 'primary'}.970`)}`;
+      case 'tertiary':
+        return '0px';
+      default:
+        return '0px';
+    }
+  },
+  underlayColor({ variant, variantColor, theme }) {
+    switch (variant) {
+      case 'primary':
+        return getColor(theme, `${variantColor || 'primary'}.600`);
```

Comment:
do this at just one place and just pass the `color` value when calling this function